### ISSUE TITLE
File path encoding bug

### ISSUE THREAD

**Issue by robmcmullen** [robmcmullen/peppy#361]
_[Trac time 20080208 072442Z]_ There is a bug with support of non-latin languages. I'm using
ru_RU.UTF-8 on my Fedora linux, so when I attempt to open a file in a
path with russian names I get the following message:
error accessing 'file:///home/h8/%C3%90%C2%97%C3%90%C2%B0%C3%90%C2%B3%
C3%91%C2%80%C3%91%C2%83%C3%90%C2%B7%C3%90%C2%BA%C3%90%C2%
B8/peppy-0.7.2.2/peppy': File not found

**Comment by robmcmullen** [comment]
_[Trac time 20080208 182828Z]_ I think I may need to use the method decode(sys.getfilesystemencoding()) on the string.  I'll try to set up a path to test on my machine here.

**Comment by robmcmullen** [comment]
_[Trac time 20080208 184854Z]_ It looks like there's also a problem in itools:

```
Traceback (most recent call last):
  File "/home/rob/src/peppy-git/peppy/actions/minibuffer.py", line 126, in OnEnter
    error = self.action.processMinibuffer(self, self.mode, text)
  File "/home/rob/src/peppy-git/peppy/mainmenu.py", line 155, in processMinibuffer
    self.frame.open(text)
  File "/home/rob/src/peppy-git/peppy/frame.py", line 670, in open
    user_url = vfs.normalize(url)
  File "/home/rob/src/peppy-git/peppy/vfs/__init__.py", line 18, in normalize
    ref = get_reference(ref)
  File "z:\src\peppy-git\peppy\vfs\itools\uri\uri.py", line 38, in get_reference
  File "/home/rob/src/peppy-mil/peppy/vfs/itools/uri/generic.py", line 699, in decode
  File "/home/rob/src/peppy-mil/peppy/vfs/itools/uri/generic.py", line 209, in __init__
UnicodeEncodeError: 'ascii' codec can't encode characters in position 4-6: ordinal not in range(128)
```

**Comment by robmcmullen** [comment]
_[Trac time 20080209 022701Z]_ 4a1cbc46590bda37d6aab6a99d196e648a355afd Fixes #361: use unicode for uri and filename access
- changed itools.vfs and support code to use unicode when passed a unicode string
- changed any use of buffer.url to unicode
- changed menu system to handle unicode
- changed _ method to return a unicode string rather than plain string
- fixed openrecent so that it serializes its filenames as utf8
- added testcases for itools unicode support
